electrumsv/wallet.py

```diff
@@ -978,17 +993,31 @@ def get_address_history(self, address: Address):
                 if tx_fee is not None:
                     flags |= TxFlags.HasFee
                 updates.append((tx_hash, data, None, flags))
-            self._datastore.tx.update_or_add(updates)
+
+                self._datastore.tx.update_or_add(updates)
 
             for tx_id in set(t[0] for t in hist):
+                has_tx_data = self._datastore.tx.have_transaction_data(tx_id)
+                is_cached = self._datastore.tx.is_cached(tx_id)
+                if has_tx_data and is_cached:
```

Comment:
This needs to be restructured. If transaction metadata is cached, then it may or may not have transaction byte data. If it has transaction byte data, then the metadata at least is guaranteed to be cached, but not the byte data. There is probably never a situation where you want to get the non-cached entry. But what you might want to do is fetch the byte data. In this case, you want to use `tx.get_transaction` to do so. If it the byte data is cached it will give it to you, otherwise it will populate it in the cache and return it.

So